src/MessagePack.UnityClient/Assets/Scripts/MessagePack/IFormatterResolver.cs

```diff
@@ -76,9 +81,17 @@ public static object GetFormatterDynamic(this IFormatterResolver resolver, Type
                 throw new ArgumentNullException(nameof(type));
             }
 
-            MethodInfo methodInfo = typeof(IFormatterResolver).GetRuntimeMethod(nameof(IFormatterResolver.GetFormatter), Type.EmptyTypes);
+            var formatterGetter = FormatterGetters.GetOrAdd(type, t =>
```

Comment:
You're allocating the closure and delegate for the second argument to `GetOrAdd` even in the (likely) event that the result is already produced and cached. Can we instead first check whether it's present and only if it isn't make those two allocations?

Reply:
Closure not allocated, but I've made changes you requested and resolving works little faster